Development

samcca opened the pull request.

jdlink9gh committed:
added project prompt to readme

jdlink9gh committed:
removed readme from dev

Commit:
file get and word selection work

Commit:
working features: directory/file checking and creation/downloading, reading the file, and selecting a random word of at least a minimum length

Commit:
working features: directory/file checking and creation/downloading, reading the file, selecting a random word of at least a minimum length, and printing of the different hangman state

Commit:
All hangman features working:
 - file check and download
 - choosing random word from list at least *input* long
 - looping through guesses until the player loses or wins
 - displays available letters for guessing and hangman state with each guess
 - error checks guesses and displays relevant error
 - notifies player if they win or lose
 - allows player to start a new game

Commit:
improved error checking and outputs

Commit:
commented argparse out

jdlink9gh committed:
Merge pull request #2 from jdlink9gh/hangman

Hangman

Commit:
Added CLI functionality using argparse

Commit:
Added read me file with code details and gameplay instructions. Dried up hangman.py

jdlink9gh committed:
Merge pull request #3 from jdlink9gh/feature/argparse

Feature/argparse

jdlink9gh commented on .gitignore:
```diff
@@ -102,3 +102,7 @@ venv.bak/
 .mypy_cache/
+
+# Project Files
+/sowpods.txt
```
what is this file? was this left over from a test?

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+    def wordPick(self):
+        # This method randomly selects a word of a minimum length from the stored list
+        # self.minlength = int(args.minlength)
+        self.word = str()  # word chosen by game
```
Isnt this already defined in the __init__? (WET)

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+        # selecting a random word from the list that has a length greater than the specified minimum
+        while len(self.word) < self.minlength:
+            self.word = random.choice(self.lines).lower()
+            self.word = self.word[:-1]
```
Can we combine these lines?

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+
+class Hangman():
+
+    def __init__(self, length): # intaking the length argument from the command line
```
make sure there are 2 spaces between code and line comment (PEP8). This could probably be fixed by using pycharms reformat file function

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+class Hangman():
+
+    def __init__(self, length): # intaking the length argument from the command line
+        self.stateHM = 0        # state of the hangman; 1-6
```
lets try and get int he habit of using "_" instead of camel casing. I know i am terrible at this, but it is the "pythonic" way. This applies to both Variables and Functions

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+        self.word = str()       # word chosen by the program
+        self.minlength = length    # setting the minimum length of the word chosen by the program to the length arg
+
+    def openFile(self):
```
camelcase

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+        for char in self.word:
+            target += ' ' + char + ' '
+
+        for char in self.word:
```
i think you can replace char on this line with "_"
you dont seem to need the actual value

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+
+        try:
+            # checking if the file exists
+            with open(filePath) as f:
```
i dont think this is the safest practice to open the file to see if it exists. Would something like :
```os.path.is_file(file_path)``` work?

Everytime we open a file, there is a risk it gets deleted or moved or something. Safer to just check if it exists

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+        pathlib.Path('./word_data/').mkdir(parents=True, exist_ok=True)
+
+        # storing the relative file path to the words.txt file
+        filePath = './word_data/words.txt'
```
camelcase

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+        if wrongGuesses > 0:  # printing second line of gallows for all states
+            print('  |       O')
+        else:
+            print('  |')
```
Good job making this part dryer. We could define ```pole = '  |' for clarity if we wanted, but that is more of a style choice

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+            guess = input("Choose a letter: ")
+            guess = guess.lower()
+
+            if guess in letters:
```
This if block is pretty confusing. We have guess, guessed, guesses, letters, and word. Might try to add more clarity to variable name. I am having trouble following the logic with the similar names. We may be able to simplify this block with clearer variable names

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+                    print(f'You already guessed "{guess}"!')
+
+                time.sleep(1)
+                self.printhangman(stateHM)
```
similar to the camel case comment, lets try to use "_" to demarcate words in our functions for ease of reading

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+
+    # the next 3 lines of code setup the CLI input using the argparse library
+    parser = argparse.ArgumentParser(description='Play a game of Hangman')
+    parser.add_argument('-l', '--minlength', type=int, metavar='', required=True,
```
we may not want to use a lowercase "L" since it looks like a 1 in the terminal, this could confuse the user. Maybe "-m"?

jdlink9gh commented on hangman.py:
```diff
@@ -0,0 +1,173 @@
+    args = parser.parse_args()
+
+    game = Hangman(length=args.minlength)
+    game.minlength = args.minlength
```
I dont think this line is necessary now that we pass it in during instantiation

jdlink9gh reviewed (changes requested):
Really good job on this code. Almost everything i have is stylistic (which will come naturally with time)

Commit:
Addressed pull request comments:
 - dried up code
 - added clarity to variable names
 - improved styling
 - removed testing file from .gitignore
 - updated read me to reflect code changes

jdlink9gh committed:
Merge pull request #5 from jdlink9gh/feature/argparse

Addressed pull request comments:
